test/model/internal_link_test.dart

```diff
@@ -451,4 +452,48 @@ void main() {
       });
     });
   });
+
+  group('narrowLink URL generation', () {
+    final streams = [
+      eg.stream(streamId: 1, name: 'check'),
+    ];
+
+    test('narrowLink should include a trailing slash in the path and before #', () async {
+      final store = await setupStore(realmUrl: realmUrl, streams: streams);
+      final message = eg.streamMessage();
+      final narrow = TopicNarrow.ofMessage(message);
+      final messageLink = narrowLink(store, narrow, nearMessageId: 12345);
+      final urlString = messageLink.toString();
+      final fragmentIndex = urlString.indexOf('#');
+      expect(fragmentIndex, greaterThan(0));
+      if (fragmentIndex > 0) {
+        final lastCharBeforeFragment = urlString[fragmentIndex - 1];
+        expect(lastCharBeforeFragment, equals('/'));
+      }
+    });
+  });
+
+  group('narrowLink Full URL generation', () {
+    final streams = [
+      eg.stream(streamId: 123, name: 'check'),
+    ];
+    test('narrowLink should generate the full URL correctly', () async {
+      final store = await setupStore(realmUrl: realmUrl, streams: streams);
+      final message = eg.streamMessage();
+      final narrow = TopicNarrow.ofMessage(message);
+      final messageLink = narrowLink(store, narrow, nearMessageId: 12345);
+      final urlString = messageLink.toString();
+      final uri = Uri.parse(urlString);
+
+      expect(uri.scheme, equals('https'));
+      expect(uri.host, equals('example.com'));
+      expect(uri.path.endsWith('/'), isTrue);
+      const expectedFragmentStart = 'narrow/stream/123-check/topic';
+      expect(uri.fragment.startsWith(expectedFragmentStart), isTrue);
+      expect(
+          urlString.startsWith('https://example.com/#narrow/stream/123-check/topic'),
+          isTrue
+      );
```

Comment:
I think this test and the one before it can be combined into one. However, looking at this more, it might be simpler to just check if `urlString` equals the URL we expect (i.e.: `'https://example.com/#narrow/stream/123-check/topic'`).

Or, even better, we can add the test to `test/widgets/action_sheet_test.dart`, as a test similar to "copies message link to clipboard". This way we get to test it end-to-end. In this case, no change is needed for `test/model/internal_link_test.dart`.